### PR TITLE
fix: performance improvement for ObservableComponents

### PR DESCRIPTION
**Pull request by eordano**

### CODE REVIEW

File: kernel/packages/decentraland-ecs/src/decentraland/Implementation.ts [hunk 1]
```diff
@@ -92,7 +97,7 @@ export class DecentralandSynchronizationSystem implements ISystem {
         const component = entity.components[componentName]
         const classId = getComponentClassId(component)
 
-        if (classId !== null) {
+        if (classId !== null && componentNameRE.test(getComponentName(component))) {
```
Comment (BrianAmadori) [inline kernel/packages/decentraland-ecs/src/decentraland/Implementation.ts:101]: Regex tests tend to be fairly slow. Are you sure this is a performance improvement? How often this path is reached?

Comment (eordano) [reply kernel/packages/decentraland-ecs/src/decentraland/Implementation.ts:101]: Regexps are fairly optimized in javascript. This avoids doing any sort of dirty comparison for components that are not the engine (any custom component is not checked). I'll do an approach with `.startsWith` to see how it looks like

Comment (eordano) [reply kernel/packages/decentraland-ecs/src/decentraland/Implementation.ts:101]: image

Comment (BrianAmadori) [reply kernel/packages/decentraland-ecs/src/decentraland/Implementation.ts:101]: Nice to know 👍

File: kernel/packages/decentraland-ecs/src/decentraland/Implementation.ts [hunk 2]
```diff
@@ -203,15 +221,21 @@ export class DecentralandSynchronizationSystem implements ISystem {
    * created component is fired immediatly after.
    */
   private disposableComponentCreated(event: DisposableComponentCreated) {
-    this.dcl.componentCreated(event.componentId, event.componentName, event.classId)
+    if (componentNameRE.test(event.componentName)) {
```
Comment (BrianAmadori) [inline kernel/packages/decentraland-ecs/src/decentraland/Implementation.ts:221]: Maybe we can use early exit in those functions instead to reduce indentation.

**Review by BrianAmadori — APPROVED** [review]

**Review by moliva — APPROVED** [review]
LGTM.

**Pull request merged**